stream_alert/shared/config.py

```diff
@@ -0,0 +1,154 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from collections import defaultdict, OrderedDict
+import json
+import os
+
+
+class ConfigError(Exception):
+    """Exception class for config file errors"""
+
+
+def parse_lambda_arn(function_arn):
+    """Extract info on the current environment from the lambda function ARN
+
+    Parses the invoked_function_arn from the given context object to get
+    the name of the currently running alias (either production or development)
+    and the name of the function.
+
+    Example:
+        arn:aws:lambda:aws-region:acct-id:function:stream_alert:production
+
+    Arguments:
+        function_arn (str): The AWS Lambda function ARN
+
+    Returns:
+        dict:
+            {
+                'region': 'region_name',
+                'account_id': 'account_id',
+                'function_name': 'function_name',
+                'qualifier': 'qualifier'
+            }
+    """
+    split_arn = function_arn.split(':')
+    return {
+        'region': split_arn[3],
+        'account_id': split_arn[4],
+        'function_name': split_arn[6],
+        'qualifier': split_arn[7] if len(split_arn) == 8 else None # optional qualifier
+    }
+
+
+def load_config(conf_dir='conf/', **kwargs):
+    """Load the configuration for StreamAlert.
+
+    All configuration files live in the `conf` directory in JSON format.
+    `sources` define a colleciton of AWS services (S3, Kinesis) supported as
+    inputs to StreamAlert, specific entities (S3 buckets, Kinesis streams),
+    and log types emitted from them.
+
+    `logs` declare the schema for the listed log types in `sources`.  Each
+    key denotes the name of the log type, and includes 'keys' used to match
+    rules to log fields.
+
+    Arguments:
+        conf_dir (str): [optional] Path from which to load the config
+
+    Keyword Arguemnts:
+        exclude (set): Names of config files or folders that should not be loaded
+        include (set): Names of specific config files to only load
+        validate (bool): Validate aspects of the config to check for user error
+    """
```

Comment:
If there are only 3 keyword arguments, can we list them explicitly instead of using `**kwargs`? I'm nearly always a fan of explicit kwargs - it makes it easier for IDEs to autocomplete, etc

Reply:
no real reason not to - I can make the change :)